docs/source/dev-guide/techspec-solver-state.md

```diff
@@ -0,0 +1,165 @@
+   constraints.
+7. `do_not_remove`: A fixed list of packages that receive special treatment by the solver due
+   to poor metadata in the early days of conda packaging. A legacy leftover.
+
+This one group _does_ change during the solver lifetime:
+
+8. `conflicting`: Specs that are suspected to be a conflict for the solver.
+
+Also, two more sources that are not obvious at first. These are not labeled as a _source_, but they
+do participate in the `specs` collection:
+
+* In new environments, packages included in the `contex.create_default_packages` list. These
+  `MatchSpec` objects are injected in each `conda create` command, so the solver will see them
+  as explicitly requested by the user (`requested`).
+* Specs added by command line modifiers. The specs here present aren't new (they are already in
+  other categories), but they might end up in the `specs` list only when a flag is added. For
+  example, `update --all` will add all the installed packages to the `specs` list, with no
+  version constraint. Without this flag, the installed packages will still end up in the `specs`
+  list, but with full constraints (`--freeze-installed` defaults for the first attempt) unless:
+    * Frozen attempt failed.
+    * `--update-specs` (or any other `UpdateModifier`) was passed, overriding `--freeze-installed`.
+
+See? It gets involved. We will also use this vocabulary to help narrow down the type of change
+being done:
+
+Types of `spec` objects:
+
+* `specs`: map of package name to its currently corresponding `MatchSpec` instance.
+* `spec`: specific instance of a `MatchSpec` object.
+* Exact or frozen `spec`: a `spec` where both the `version` and `build` fields are constrained
+  with `==` operators (exact match).
+* Fully constrained or tight `spec`: a `spec` where both `version` and `build` are populated,
+  but not necessarily with equality operators. It can also be inequalities (`>`, `<`, etc.) and
+  fuzzy matches (`*something*`).
+* Version-only `spec`: a `spec` where _only_ the `version` field is populated. The `build`
+  is not.
+* Name-only, bare, or unconstrained `spec`: a `spec` with no `version` or `build ` fields. Just
+  the name of the package.
+* Targeted `spec`: a `spec` with the `target` field populated. Extracted from the comments in
+  the solver logic:
+    > `target` is a reference to the package currently existing in the environment. Setting
+    > `target` instructs the solver to not disturb that package if it's not necessary. If the
+    > spec.name is being modified by inclusion in `specs_to_add`, we don't set `target`, since we
+    > *want* the solver to modify/update that package.
+    >
+    > TL;DR: when working with `MatchSpec` objects,
+    >  - to minimize the version change, set `MatchSpec(name=name, target=prec.dist_str())`
+    >  - to freeze the package, set all the components of `MatchSpec` individually
+* if the `spec` object does not have an adjective, it should be assumed it's being added to the
```

Comment:
Something seems to be going wrong with the Markdown formatting here. I'm not sure if it affects the output. Maybe add a line beneath the blockquote?